### DISCUSSION TITLE
Prolem while cellconstructor instalation on a cluster

### DISCUSSION THREAD

**Discussion by szkudlarek-piotr** [mesonepigreco/CellConstructor#71] [General]
Hi,
I am trying to install cellconstructor on the machine of my team. However, when I am trying to execute pip install cellconstructor, I am getting following error:

![image](https://github.com/SSCHAcode/CellConstructor/assets/127056550/44449738-cea6-4b09-bc8b-6869fe747169)
I don't exactly know what I should do in this case. Should I install new Python package and install older version of Numpy there?
Thank you in advance for help,
Piotr Szkudlarek

**Comment by diegomartinez2** [comment]
Hi,
I see you are not using anaconda. Did you installed the "python-dev"?
`sudo apt install python-dev`

**Comment by mesonepigreco** [comment]
Hi, an alternative as Diego mentioned, is to install Anaconda, which allows you to install codes without administration rights:

Download Anaconda for your machine:
https://www.anaconda.com/
Once you have installed it, create a new virtual environment and install cellconstructor
```
conda create -n cellconstructor -c conda-forge python=3.10 gfortran libblas lapack openmpi julia openmpi-mpicc pip numpy scipy spglib 
pip install cellconstructor -U
```

This way, the code should work out of the box without contaminating your environment. This is the easiest way on a cluster in which you may have difficulties to install all the requirements

**Reply by szkudlarek-piotr** [reply]
Thank you for your fast response, Lorenzo.
I tried the solution you mentioned. However, cellconstructor installer still seems to be dissatisied with the deprecated numpy command. I am attachoing a txt file that contains all the output from the installer. To reducre the size, I replaced warnings  about unused variable with the proper comment.
I forgot to write - I am operating on RedHat system.
Thank you once again,
Piotr
[cellconstructor_errors.txt](https://github.com/SSCHAcode/CellConstructor/files/12000632/cellconstructor_errors.txt)

**Reply by mesonepigreco** [reply]
Hi, as mentioned in the error, you are still using the base environment and compiling with the standard data.

Try with
 conda activate cellconstructor
 Before installing cellconstructor with the pip command.

**Comment by diegomartinez2** [comment]
```
      /usr/bin/ld: cannot find -llapack
      /usr/bin/ld: cannot find -lblas
      collect2: error: ld returned 1 exit status
```
Looks like the problem is in the FORTRAN compilation due to the LAPACK and BLAS libraries.
